File: apps/mobile-flutter/lib/app/contracts/audio_contract.dart
class AudioContract {
  static const String recorderChannel = 'voice2text/recorder';
  static const String transcriptionEventChannel =
      'voice2text/transcription_events';

  static const String audioDirName = 'audios';
  static const String recordingDirName = 'recordings';
  static const String recordingInProgressDirName = 'in-progress';
  static const String recordingCompleteDirName = 'complete';
  static const String recordingJournalDirName = 'journals';
  static const String recordingJournalSuffix = '.journal.json';
  static const String importDirName = 'imports';
  static const String importInProgressDirName = 'in-progress';
  static const String importCompleteDirName = 'complete';
  static const String transcriptExportDirName = 'transcript-exports';
  static const String exportDirName = 'exports';
  static const String recordingExtension = 'm4a';
  static const String recordingStagingExtension = 'm4a.partial';

  static const int recordingConsentVersion = 1;
  static const int minimumStorageReserveBytes = 512 * 1024 * 1024;
  static const int storageCheckIntervalMs = 15000;
  static const int maximumImportedMediaBytes = 2 * 1024 * 1024 * 1024;
  static const int maximumImportedDurationMs = 4 * 60 * 60 * 1000;

  static const int sampleRateHz = 16000;
  static const int bitRate = 64000;
  static const int channelCount = 1;

  static const String containerFormat = 'mpeg4';
  static const String codec = 'aac';

  // Keep this up-to-date with Android native implementation.
  static const String compatibilityNote =
      'Standard recorder output is m4a/aac @16kHz mono.';
}

File: apps/mobile-flutter/android/app/src/main/kotlin/com/voice2text/app/contracts/AudioContract.kt
package com.voice2text.app.contracts

object AudioContract {
    const val RECORDER_CHANNEL = "voice2text/recorder"
    const val TRANSCRIPTION_EVENT_CHANNEL = "voice2text/transcription_events"

    const val AUDIO_DIR_NAME = "audios"
    const val RECORDING_DIR_NAME = "recordings"
    const val RECORDING_IN_PROGRESS_DIR_NAME = "in-progress"
    const val RECORDING_COMPLETE_DIR_NAME = "complete"
    const val RECORDING_JOURNAL_DIR_NAME = "journals"
    const val RECORDING_JOURNAL_SUFFIX = ".journal.json"
    const val IMPORT_DIR_NAME = "imports"
    const val IMPORT_IN_PROGRESS_DIR_NAME = "in-progress"
    const val IMPORT_COMPLETE_DIR_NAME = "complete"
    const val TRANSCRIPT_EXPORT_DIR_NAME = "transcript-exports"
    const val EXPORT_DIR_NAME = "exports"
    const val RECORDING_EXTENSION = "m4a"
    const val RECORDING_STAGING_EXTENSION = "m4a.partial"

    const val RECORDING_NOTIFICATION_CHANNEL_ID = "audio_recording"
    const val RECORDING_NOTIFICATION_ID = 4101
    const val RECORDING_CONSENT_VERSION = 1

    const val MINIMUM_STORAGE_RESERVE_BYTES = 512L * 1024L * 1024L
    const val STORAGE_CHECK_INTERVAL_MS = 15_000L
    const val MAXIMUM_IMPORTED_MEDIA_BYTES = 2L * 1024L * 1024L * 1024L
    const val MAXIMUM_IMPORTED_DURATION_MS = 4L * 60L * 60L * 1000L

    const val SAMPLE_RATE_HZ = 16000
    const val BIT_RATE = 64000
    const val CHANNEL_COUNT = 1

    const val CONTAINER_FORMAT = "mpeg4"
    const val CODEC = "aac"
}
